### DISCUSSION TITLE
Small additions to automations

### DISCUSSION THREAD

**Discussion by benaunt** [Budibase/budibase#11282] [Ideas]
Hi,

Have a few ideas to improve the automation module.

Idea 1:
We developed a rather complicated workflow that we ended up not needing at the moment, rather than delete the workflow. It would be great if they can just be placed in a disabled state.

In this state it would stop any triggering to happen when in preview or published. Some considerations will need to be made with app triggered workflows and how to manage the interaction if the workflow is disabled.

Idea 2:
It would be good if the workflow can have a free text description applied and if each action can have a comment stored against them to help with internal documentation of the process.

Idea 3:
For app triggered workflows, it would be nice to be able to see from the workflow screen all components within the UI that are triggering the workflow.

Thanks

**Comment by JCV67** [comment]
4 - the ability of having tree conditional options i.e On the same automation having the possibility of different endings according to different conditions.

**Reply by benaunt** [reply]
Want to stress this one, the current condition action is not sufficient for any complex workflow that needs to either have pathways for both a true and false response to the condition or more complex switch based pathways with multiple paths depending on a value.

**Comment by melohagan** [comment]
Great suggestions!

A little workaround for *Idea 1* would be to add a conditional block at the top that is always false, e.g.

![Screenshot 2023-08-01 at 15 46 43](https://github.com/Budibase/budibase/assets/101575380/4bfb8121-fa37-4b2e-828a-ad6cae02b823)

**Comment by benaunt** [comment]
I have some additional ones after building more workflows.

- Ability to re-arrange workflow actions by dragging them up or down the editor, doing so should automatically adjust any actions that reference an output of the action being moved (step1.value, step2.value etc). It should also warn if an action is moved above a parent action who's values it references, but should not restrict the movement from happening
- Improved awareness of output values, at the moment the UI allows you to reference an actions or triggers value by using the UI but only goes as deep as: {{ trigger.fields }} or {{ steps.1.result }}, to get to the actual values you need to manually type the fields. 
- When viewing the automation list the user should be able to resize the width of this window or the UI should displays the cut of names when hovering over them 
- Saving and version history of workflows: at the moment the workflow saves as soon as an edit is done. I believe it may be useful to allow a manually triggered save that allows version notes to be added. There should also be a history with rollback that logs all automated saves and displays any manual saves.
